### PR TITLE
MongoDB Atlas Mapping

### PR DESCRIPTION
**Pull request by eepstain**
## Contributing to Cortex XSOAR Content
Make sure to register your contribution by filling the [contribution registration form](https://forms.gle/XDfxU4E61ZwEESSMA)

**The Pull Request will be reviewed only after the contribution registration form is filled.**

## Status
- [x] In Progress
- [ ] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: [link to the issue](https://jira-dc.paloaltonetworks.com/browse/CIAC-10906)

## Description
A few sentences describing the overall goals of the pull request's commits.

## Must have
- [ ] Tests
- [ ] Documentation

### CODE REVIEW

File: Packs/MongoDBAtlas/README.md [hunk 1]
```diff
@@ -1,11 +1,28 @@
...
+* Alerts.
+
+### Supported Timestamp Formats:
+This pack provides an automated timestamp ingestion using the available integration for fetching logs.
+
+### MongoDB Atlas Event Collector:
+To enable the MongoDB Atlas Event Collector, follow these steps:
+1. Go to **Settings** &rarr; **Configurations** &rarr; **Automation & Feed Integrations**.
+2. In the search bar, type **MongoDB Atlas**.
+3. At the right-corner, click on **+ Add instance**.
```
Comment (ShirleyDenkberg) [inline Packs/MongoDBAtlas/README.md:21]:
```suggestion
3. At the right-corner, click **+ Add instance**.
```

File: Packs/MongoDBAtlas/ReleaseNotes/1_0_1.md [hunk 2]
```diff
@@ -0,0 +1,6 @@
+
+#### Modeling Rules
+
+##### New: MongoDB Atlas Modeling Rule
+
+<~XSIAM> Created an XDM fields mapping for MongoDB Atlas logs (Available from Cortex XSIAM v2.4).</~XSIAM>
```
Comment (ShirleyDenkberg) [inline Packs/MongoDBAtlas/ReleaseNotes/1_0_1.md:6]: Since this is only for XSIAM is this tag necessary?

**Comment by ShirleyDenkberg** [comment]
@eepstain Doc review completed.

File: Packs/MongoDBAtlas/ReleaseNotes/1_0_1.md (hunk 2 above)
Comment (eepstain) [reply Packs/MongoDBAtlas/ReleaseNotes/1_0_1.md:6]: Doesn't really do any harm if kept.
If in the future an XSOAR support will be added, it's for the better.

File: Packs/MongoDBAtlas/ModelingRules/MongoDBAtlas/MongoDBAtlas.xif [hunk 3]
```diff
@@ -0,0 +1,41 @@
+[MODEL: dataset="mongodb_atlas_raw"]
+filter source_log_type = "alerts"
+| alter
+	xdm.event.type = source_log_type,
+	xdm.alert.description = acknowledgementComment,
+	xdm.source.user.username = acknowledgingUsername,
```
Comment (OBavly) [inline Packs/MongoDBAtlas/ModelingRules/MongoDBAtlas/MongoDBAtlas.xif:6]: I'm not sure this is the field I would map it to - To my understanding, Acknowledging Username is the username of the user who worked on the alert on the SOC side, while the source.user.username is likely associated with the username that the alert was triggered on

File: Packs/MongoDBAtlas/ModelingRules/MongoDBAtlas/MongoDBAtlas.xif [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+	xdm.alert.original_alert_id = coalesce(alertConfigId, id),
+	xdm.event.original_event_type = eventTypeName,
+	xdm.source.user.groups = arraycreate(groupId),
+	xdm.observer.unique_identifier = arraystring(arraydistinct(arraymap(links -> [], "@element" -> href)), "|");
+
+filter source_log_type = "events"
+| alter
+	get_linkToDetails = arraycreate(linkToDetails),
+	get_links_href = arraydistinct(arraymap(links -> [], "@element" -> href)),
+	get_username = arraystring(arraycreate(acknowledgingUsername, username), "|")
```
Comment (OBavly) [inline Packs/MongoDBAtlas/ModelingRules/MongoDBAtlas/MongoDBAtlas.xif:16]: Same issue with the username here

Comment (eepstain) [reply Packs/MongoDBAtlas/ModelingRules/MongoDBAtlas/MongoDBAtlas.xif:16]: Removed the mapping of acknowledgingUsername for alerts.

**Review by OBavly — APPROVED** [review]

**Pull request merged**